usr/share/rear/layout/save/GNU/Linux/200_partition_layout.sh

```diff
@@ -30,14 +30,21 @@ extract_partitions() {
 
     declare path sysfs_path
     if [[ ${#sysfs_paths[@]} -eq 0 ]] ; then
-        ### try to find partitions like /dev/mapper/datalun1p1
         if [[ ${device/mapper//} != ${device} ]] ; then
-            for path in ${device}p[0-9]* ${device}[0-9] ${device}-part* ${device}_part*; do
-                sysfs_path=$(get_sysfs_name $path)
-                if [[ "$sysfs_path" ]] && [[ -e "/sys/block/$sysfs_path" ]] ; then
-                    sysfs_paths=( "${sysfs_paths[@]}" "/sys/block/$sysfs_path" )
-                fi
-            done
+            ### /sys/block/dm-X/holders directory contains partition name in dm-X format.
+            if [ -d /sys/block/$sysfs_name/holders ]; then
+                sysfs_paths=( /sys/block/$sysfs_name/holders/* )
+            else
+                ### if the holders directory does not exisits 
+                ### failback to partition name guessing method.
+                ### try to find partitions like /dev/mapper/datalun1p1
+                for path in ${device}p[0-9]* ${device}[0-9]* ${device}-part* ${device}_part*; do
```

Comment:
Is the asterisk at the end of `${device}[0-9]*` intended since it is missing in original line ? 
I'm just asking without knowing anything further about this code ;-)

Reply:
@jsmeix  "awk eyes" points this "missing trailing *" in this comment https://github.com/rear/rear/issues/1766#issuecomment-378602550

Reply:
@schabrolles 
what are "awk eyes"?
( I am not a regular `awk` user - for my needs `sed` is sufficient ;-)

Reply:
@jsmeix, typo ... => "hawk eyes" ;-)